Magicseaweed sensor documentation

jcconnell opened the pull request:
**Description:**

Documentation for the magicseaweed sensor.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#15132

## Checklist:

- [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [x] The documentation follow the [standards][standards].

jcconnell committed:
Magicseaweed sensor documentation

frenck commented on source/_components/sensor.magicseaweed.markdown:
```diff
@@ -0,0 +1,66 @@
+logo: magicseaweed.png
+ha_category: Sensor
+featured: false
+ha_release: "0.72"
```
0.72 is already out. You might want to up this to 0.73 :arrow_up:

frenck commented on source/_components/sensor.magicseaweed.markdown:
````diff
@@ -0,0 +1,66 @@
+      - max_breaking_swell
+```
+
+Configuration variables:
````
Please use the configuration tags, for more information please see:
https://home-assistant.io/developers/documentation/create_page/#configuration

frenck commented on source/_components/sensor.magicseaweed.markdown:
```diff
@@ -0,0 +1,66 @@
+# Example configuration.yaml entry
+sensor:
+  - platform: magicseaweed
+    name: Hurricane Spot
```
We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

frenck commented on source/_components/sensor.magicseaweed.markdown:
```diff
@@ -0,0 +1,66 @@
+comments: false
+sharing: true
+footer: true
+logo: magicseaweed.png
```
This logo is missing in the repository, maybe you forgot to add it?

jcconnell replied:
Yes, I did forget to add it. I'm still new at this. What's the procedure for adding it / where does it go?

jcconnell replied:
Just added. A transparent background image was not available. According to their API documentation, they request that the logo I uploaded be used.

frenck reviewed (dismissed):
Hi @jcconnell,

I've reviewed your documentation PR. Could you take a look at the comments? 👍

jcconnell committed:
Making @frenck requested changes.

Not sure I did this correctly but I followed the docs: https://www.home-assistant.io/developers/documentation/create_page/#configuration

jcconnell committed:
Add Magicseaweed logo

Transparent background version not available. API Documentation requests that this logo be used.

jcconnell committed:
Changed logo name.

jcconnell committed:
Remove incorrectly named logo.

frenck commented:
@jcconnell  Thanks for the updates, nevertheless, the builds are failing. 
Could you take a look? Thx! 👍

jcconnell commented:
Hello @frenck, I apologize for the delay. I was asked to write a library to supplement this sensor. Now that it's finished, I'm revisiting this error with the documentation. Jekyll is reporting an error on line 9 column 25 regarding mapping values on sensor.magicseaweed.markdown. I've reviewed the file and searched for similar problems, but I'm still unable to find the issue. Do you see anything I might be missing?

jcconnell committed:
Correct mapping error line 9 column 25 of YAML

jcconnell committed:
HA Release Update & removed irrelevant config

jcconnell committed:
Attempting to fix Jekyll errors

jcconnell commented:
Ok, it seems that I have addressed all the problems with my file, sensor.magicseaweed.markdown. It is now complaining about a filename for MQTT logging:

`Invalid filename 'components/mqtt/#logging.html'. Deployed filenames cannot contain # or ? characters`

frenck commented:
@jcconnell  Rebase this PR onto `origin/next`, that will probably do the trick.

jcconnell commented:
Sorry, I'm not very familiar with this workflow. What would be the process to rebase this onto origin/next? Can it be completed through a web browser?

ericnewton76 commented:
@jcconnell i'll walk you through this via teamviewer

MimbaMonkeyHouse commented:
Did anyone have any luck with this? Tell us if help is needed...

MartinHjelmare commented on source/_components/sensor.magicseaweed.markdown:
```diff
@@ -0,0 +1,95 @@
+    12AM:
+      description: Display forecast for 12AM.
+spot_id:
+  description: Surf spot ID to monitor surf forecast of.Details for getting spot id available at [Magicseaweed](https://magicseaweed.com/developer/forecast-api)
```
There's a missing space after the period before "Details". But maybe rewrite the description somewhat? It reads a bit weird when ending with the preposition in the first sentence.

MartinHjelmare commented on source/_components/sensor.magicseaweed.markdown:
```diff
@@ -0,0 +1,95 @@
+  required: true
+  type: string
+monitored_conditions:
+  description: Conditions to display in the frontend.
```
This isn't dependent on the frontend per se, but decides what entities to create, ie what data to make available.

jcconnell committed:
Add @MartinHjelmare requested changes.

MartinHjelmare commented:
Can be merged when build passes.

jcconnell commented:
Yes! Thanks everyone for the help and patience.
